### PR TITLE
nft royalty edge cases

### PR DESCRIPTION
**Pull request by trepca**
### Purpose:
additional tests for nft royalty

### CODE REVIEW

File: tests/wallet/nft_wallet/test_nft_1_offers.py [hunk 1]
```diff
@@ -1272,7 +1272,17 @@ async def get_trade_and_status(trade_manager: Any, trade: Any) -> TradeStatus:
...
+    [
+        (True, (200, 500, 500)),
+        (False, (200, 500, 500)),
+        (False, (0, 0, 0)),
+        (False, (9999, 8000, 5000)),
+        (False, (20005, 30005, 40001)),
+        (False, (20000, 30000, 50000)),
+        (False, (65000, 65534, 65535)),
+        (False, (10000, 10001, 10005)),
+        (False, (100000, 10001, 10005)),
```
Comment (Quexington) [inline tests/wallet/nft_wallet/test_nft_1_offers.py:1284]: With the parameter above, this is the equivalent of 12 new tests, are they all strictly necessary?

File: tests/wallet/nft_wallet/test_nft_1_offers.py [hunk 2]
```diff
@@ -1316,12 +1326,23 @@ async def test_complex_nft_offer(
     for i in range(0, 2):
         await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph_maker))
         await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph_taker))
-    await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph_taker))
+    if not forwards_compat:
+        for i in range(9):
```
Comment (Quexington) [inline tests/wallet/nft_wallet/test_nft_1_offers.py:1330]: Why do we need 9 blocks now?  The amounts still seem pretty small yeah?

Comment (trepca) [reply tests/wallet/nft_wallet/test_nft_1_offers.py:1330]: because >100% royalty gets expensive

File: tests/wallet/nft_wallet/test_nft_1_offers.py (hunk 1 above)
Comment (trepca) [reply tests/wallet/nft_wallet/test_nft_1_offers.py:1284]: forgot to clean up, thanks

File: tests/wallet/nft_wallet/test_nft_1_offers.py [hunk 3]
```diff
@@ -1272,7 +1272,14 @@ async def get_trade_and_status(trade_manager: Any, trade: Any) -> TradeStatus:
...
 @pytest.mark.parametrize(
     "forwards_compat,royalty_pts",
-    [(True, (200, 500, 500)), (False, (200, 500, 500)), (False, (0, 0, 0))],
+    [
+        (True, (200, 500, 500)),
+        (False, (200, 500, 500)),
+        (False, (0, 0, 0)), # test that we can have 0 royalty
+        (False, (65000, 65534, 65535)), # test that we can reach max royalty
+        (False, (10000, 10001, 10005)), # tests 100% royalty is not allowed
+        (False, (100000, 10001, 10005)), # 1000% should work
```
Comment (Quexington) [inline tests/wallet/nft_wallet/test_nft_1_offers.py:1281]:
```suggestion
        (False, (100000, 10001, 10005)), # 1000% shouldn't work
```

**Review by Quexington — DISMISSED** [review]

**Review by Quexington — DISMISSED** [review]

**Review by paninaro — DISMISSED** [review]
lgtm

**Review by paninaro — APPROVED** [review]

**Review by Quexington — APPROVED** [review]
